platform/migration/platform-migratable-api/src/main/java/org/codice/ddf/migration/ExportMigrationContext.java

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.migration;
+
+import java.nio.file.Path;
+import java.nio.file.PathMatcher;
+import java.util.Optional;
+import java.util.function.BiPredicate;
+import java.util.stream.Stream;
+
+/**
+ * The export migration context provides functionality for creating new migration entries and
+ * tracking exported migration entries for a given migratable while processing an export migration
+ * operation.
+ * <p>
+ * <b>
+ * This code is experimental. While this interface is functional
+ * and tested, it may change or be removed in a future version of the
+ * library.
+ * </b>
+ * </p>
+ */
+public interface ExportMigrationContext extends MigrationContext {
+    /**
+     * Creates or retrieves (if already created) a migration entry referenced from the specified system
+     * property to be exported by the corresponding migratable.
+     * <p>
+     * An error will automatically be recorded with the associated migration report if the system
+     * property is not defined or its value is blank.
+     *
+     * @param name the name of the system property referencing a migration entry to create or retrieve
+     * @return a new migration entry or the existing one if already created for the migration entry
+     * referenced from the specified system property value or empty if the property is not defined or
+     * its value is blank
+     * @throws IllegalArgumentException if <code>name</code> is <code>null</code>
+     */
+    public default Optional<ExportMigrationEntry> getSystemPropertyReferencedEntry(String name) {
+        return getSystemPropertyReferencedEntry(name, (r, v) -> true);
+    }
+
+    /**
+     * Creates or retrieves (if already created) a migration entry referenced from the specified system
+     * property to be exported by the corresponding migratable.
+     * <p>
+     * The provided predicate will be invoked with the associated migration report and the property
+     * value (may be <code>null</code> if not defined). In case the property is not defined or is
+     * blank, an error will be automatically recorded unless the predicate returns <code>false</code>.
```

Comment:
I agree with @Lambeaux that this comment and behavior is a little confusing. Is the actual behavior something like this?

- If the predicate returns `true`, a migration entry is created and returned
- If the predicate returns `false`, no migration entry is created and no error is recorded
- If the property is `null` or blank and the predicate returns `true`, no migration entry will be created and an error will be recorded
- If the property is `null` or blank and the predicate returns `false`, no migration entry will be created and no error will be recorded

Could we simplify this a little and say that if the property is `null` or blank, an error will always be recorded and the predicate will not be called?